checker/src/main/java/alluxio/checker/SparkIntegrationChecker.java

```diff
@@ -0,0 +1,265 @@
+
+  @Parameter(names = {"-p", "--partition"}, description = "The user defined partition number")
+  private static int sPartition;
+
+  private static final Logger LOG = LoggerFactory.getLogger(SparkIntegrationChecker.class);
+  private static List<Tuple2<Integer, String>> sSparkJobResult;
+  private static boolean sAlluxioHAMode = false;
+  private static String sZookeeperAddress = "";
+
+  /**
+   * Implements check integration of Spark with Alluxio.
+   *
+   * @param sc current JavaSparkContext
+   * @return 0 succeed, 1 fail to recognize Alluxio classes,
+   * 2 fail to recognize Alluxio filesystem, 3 fail to support Alluxio HA mode
+   */
+  private static int checkIntegration(JavaSparkContext sc) {
+    // Checks whether Spark driver can recognize Alluxio classes and filesystem
+    int driverStatus = getStatus();
+    if (driverStatus != 0) {
+      return driverStatus;
+    }
+
+    // Supports Alluxio high availability mode
+    if (alluxio.Configuration.getBoolean(PropertyKey.ZOOKEEPER_ENABLED)) {
+      sAlluxioHAMode = true;
+      try {
+        sZookeeperAddress = alluxio.Configuration.get(PropertyKey.ZOOKEEPER_ADDRESS);
+      } catch (RuntimeException e) {
+        LOG.error("Zookeeper address has not been set:" + e);
+        return 3;
+      }
+    }
+
+    // Runs a Spark job to check Spark executors configuration
+    return runOperations(sc);
+  }
+
+  /**
+   * Spark job to check whether Spark executors can recognize Alluxio filesystem.
+   *
+   * @param sc current JavaSparkContext
+   * @return 0 on success, 1 on failures
+   */
+  private static int runOperations(JavaSparkContext sc) {
+    // Generates a list of integer for testing
+    List<Integer> nums = IntStream.rangeClosed(1, sPartition).boxed().collect(Collectors.toList());
+
+    JavaRDD<Integer> dataSet = sc.parallelize(nums, sPartition);
+
+    // Runs a Spark job to check whether Spark executors can recognize Alluxio
+    JavaPairRDD<Integer, String> extractedStatus = dataSet
+        .mapToPair(s -> new Tuple2<>(getStatus(), getAddress(s)));
+
+    // Merges the IP addresses that can/cannot recognize Alluxio
+    JavaPairRDD<Integer, String> mergeStatus = extractedStatus.reduceByKey((a, b) -> merge(a, b));
+
+    sSparkJobResult = mergeStatus.collect();
+
+    // If one executor cannot recognize Alluxio, something wrong has happened
+    boolean canRecognizeClass = true;
+    boolean canRecognizeFS = true;
+    for (Tuple2<Integer, String> op : sSparkJobResult) {
+      if (op._1() == 1) {
+        canRecognizeClass = false;
+      }
+      if (op._1() == 2) {
+        canRecognizeFS = false;
+      }
+    }
+    return canRecognizeClass ? (canRecognizeFS ? 0 : 2) : 1;
+  }
+
+  /**
+   * Checks if this Spark driver or executors can recognize Alluxio classes and filesystem.
+   *
+   * @return 0 succeed, 1 fail to recognize Alluxio classes,
+   * 2 fail to recognize Alluxio filesystem
+   */
+  private static int getStatus() {
```

Comment:
as I mentioned, define status in enum rather than using int to encode the state, not informative and not type-safe